### ISSUE TITLE
Travel History Observation: Support nested "component" coding elements

### ISSUE THREAD

**Issue by kmarx** [drajer-health/eCRNow#623]
Our R4 FHIR server emits Observation travel history coding as nested `component`s. This isn't currently supported in eCRNow so the travel history isn't emitted correctly in the eICR. We're following the HL7 specification for this. E.g., https://build.fhir.org/ig/HL7/case-reporting/Observation-observation-travel-history-eve-everywoman-201801.json.html.

We're also wondering why the effectivePeriod gets flattened into a single timestamp instead of start/end.

See attached full payloads.
[travel-history (1).zip](https://github.com/drajer-health/eCRNow/files/13549931/travel-history.1.zip)

**For example, we emit JSON like this:**

```
...
"effectivePeriod": {
    "start": "2023-08-07T00:00:00-07:00",
    "end": "2023-08-20T00:00:00-07:00"
  },
"component" : [
{
  "code" : {
    "coding" : [
      {
        "system" : http://terminology.hl7.org/CodeSystem/v3-ParticipationType,
        "code" : "LOC",
        "display" : "Location"
      }
    ]
  },
  "valueCodeableConcept" : {
    "coding" : [
      {
        "system" : "urn:iso:std:iso:3166",
        "code" : "FJI",
        "display" : "Fiji"
      }
    ]
  }
}, 
```

**But the eICR contains:**

```
<entry>
    <act classCode="ACT" moodCode="EVN">
        <templateId root="2.16.840.1.113883.10.20.15.2.3.1" extension="2016-12-01"/>
        <id root="d95c6804-f977-45be-85ab-1f33a8f4cff5"/>
        <code code="420008001" codeSystem="2.16.840.1.113883.6.96" codeSystemName="SNOMED-CT" displayName="Travel"/>
        <text>Travel History-</text>
        <statusCode code="completed"/>
        <effectiveTime value="20230807000000-0700"/>
    </act>
</entry>
```

**Comment by nbashyam** [comment]
Target for 3.1.3

**Comment by VenkatSaiReddyTeegala** [comment]
Resolved as part of 3.1.4
